### PR TITLE
FIREFLY-1260: Remove multiple copies of QueryMOS

### PR DESCRIPTION
**Pull request by loitly**
https://jira.ipac.caltech.edu/browse/FIREFLY-1260

Cleanup code.  Updated QueryMOS.
- also, fixed a bug where the database file exists but, there's no data in it this may be caused by a bad tomcat shutdown.

More changes and test URL here: https://github.com/IPAC-SW/irsa-ife/pull/268

### CODE REVIEW

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/mos/QueryMOS.java [hunk 1]
```diff
@@ -172,6 +188,18 @@ private URL createURL(MOSRequest req, boolean forGator) throws EndUserException,
         return new URL(url);
     }
 
+    private static String getObsDate(String obsDateParam) {
+        SimpleDateFormat receivedFormat = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ssX"); // sent by client, also ensures parsing as UTC
+        SimpleDateFormat desiredFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss"); // expected by MOST
+        try {
+            Date date = receivedFormat.parse(obsDateParam.trim());
+            return desiredFormat.format(date);
+        }
```
Comment (loitly) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/mos/QueryMOS.java:197]: @wmiipac when trying to fix this for UTC format, you broke date string without time. e.g.
```
Date date = r.parse("2023-07-17");
java.text.ParseException: Unparseable date: "2023-07-17"
	at java.base/java.text.DateFormat.parse(DateFormat.java:399)
```

Also, parsing a UTC date, then converting it back to a locale dependent format may be problematic if the server's locale is not UTC.  You should fix these.

Comment (wmiipac) [reply src/firefly/java/edu/caltech/ipac/firefly/server/query/mos/QueryMOS.java:197]: May be not using TimeRangePanel makes more sense for Precovery/SSO searches? SHA is the first one try use it

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/EmbeddedDbProcessor.java [hunk 2]
```diff
@@ -157,7 +157,7 @@ public DataGroupPart getData(ServerRequest request) throws DataAccessException {
             boolean dbFileCreated = false;
             File dbFile = getDbFile(treq);
             jobExecIf(v -> v.progress(10, "fetching data..."));
-            if (!dbFile.exists()) {
+            if (!dbFile.exists() || !EmbeddedDbUtil.hasTable(treq, dbFile, MAIN_DB_TBL)) {
```
Comment (wmiipac) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/EmbeddedDbProcessor.java:160]: was this caused MOST search table not showing in the result panel?

**Review by wmiipac — APPROVED** [review]
Looks good!

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/EmbeddedDbProcessor.java (hunk 2 above)
Comment (loitly) [reply src/firefly/java/edu/caltech/ipac/firefly/server/query/EmbeddedDbProcessor.java:160]: No.  This is the cause of `DATA not found` exception we see after redeploying an app multiple times.  You normally see this during development.  Not related to MOS.

**Pull request merged**